controlplane/kubeadm/internal/cluster.go

```diff
@@ -345,6 +274,32 @@ func (c *cluster) controlPlaneIsHealthy(ctx context.Context) (healthCheckResult,
 	return response, nil
 }
 
+func (c *cluster) removeMemberForNode(ctx context.Context, nodeName string) error {
+	tlsConfig, err := c.generateEtcdTLSClientBundle()
+	if err != nil {
+		return err
+	}
+
+	// Create the etcd client for the etcd Pod scheduled on the Node
+	etcdClient, err := c.getEtcdClientForNode(nodeName, tlsConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to create etcd client")
+	}
+
+	// List etcd members. This checks that the member is healthy, because the request goes through consensus.
+	members, err := etcdClient.Members(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to list etcd members using etcd client")
+	}
+	member := etcdutil.MemberForName(members, nodeName)
```

Comment:
shall we check if member==nil ? for the case that etcd member removed but somehow machine object is not deleted successfully.

Reply:
Possibly, will need to reconcile the best place to handle that scenario to avoid accidentally passing a health check when we shouldn't.

Thanks for bringing this up :)

Reply:
I don't believe this will still be an issue with the annotation-based approach for upgrade re-entrancy now.